### ISSUE TITLE
CA1822 false positive for C#14 extension method

### ISSUE THREAD

**Issue by znakeeye** [dotnet/roslyn-analyzers#7646]
Creat a new project and target `.NET 10` etc.
```xml
<Project>
  <PropertyGroup>
    <Nullable>enable</Nullable>
    <LangVersion>preview</LangVersion>
    <TargetFramework>net10.0</TargetFramework>
    <TreatWarningsAsErrors>true</TreatWarningsAsErrors>
    <EnforceCodeStyleInBuild>true</EnforceCodeStyleInBuild>
    <AnalysisMode>All</AnalysisMode>
    <CodeAnalysisTreatWarningsAsErrors>true</CodeAnalysisTreatWarningsAsErrors>
  </PropertyGroup>
</Project>
```

VS incorrectly suggests that the method below can be **static**. It cannot.
```
internal static class StringExtensions
{
    extension(string instance)
    {
        public bool L33t() => instance.Length == 1338;
    }
}
```

![Image](https://github.com/user-attachments/assets/be67c6ea-3a36-4e78-b277-7d220cde854a)

Relates to extensions (test plan https://github.com/dotnet/roslyn/issues/76130)

**Comment by CyrusNajmabadi** [comment]
Hitting this here: https://github.com/dotnet/roslyn/pull/79550

**Comment by Frulfump** [comment]
> Fixed by [#7739](https://github.com/dotnet/roslyn-analyzers/pull/7739).

Which release will include the fix?

**Comment by jjonescz** [comment]
If I understand correctly, these analyzers ship in the SDK, so that would be rc.1 probably. Or you can maybe install this manually? https://dev.azure.com/dnceng/public/_artifacts/feed/dotnet10/NuGet/Microsoft.CodeAnalysis.NetAnalyzers/overview/10.0.0-preview.25375.1
